finality-aleph/src/sync/handler.rs

```diff
@@ -265,22 +240,34 @@ impl<I: PeerId, J: Justification, CS: ChainStatus<J>, V: Verifier<J>, F: Finaliz
             .verifier
             .verify(state.top_justification())
             .map_err(Verifier)?;
+        let remote_top_number = remote_top.header().id().number();
         let local_top = self.chain_status.top_finalized().map_err(ChainStatus)?;
-        match local_top
-            .header()
-            .id()
-            .number()
-            .checked_sub(remote_top.header().id().number())
-        {
-            // If we are just one behind then normal broadcasts should remedy the situation.
-            Some(0..=1) => Ok(SyncActions::noop()),
-            Some(2..=MAX_SMALL_JUSTIFICATION_BATCH) => Ok(SyncActions::response(
-                self.small_justification_batch_from(remote_top.header().id())?,
+        let local_top_number = local_top.header().id().number();
+        let remote_session = session_id_from_block_num(remote_top_number, self.period);
+        let local_session = session_id_from_block_num(local_top_number, self.period);
+        match local_session.0.checked_sub(remote_session.0) {
+            // remote session number larger than ours, nothing to do here
```

Comment:
Wait, that sounds wrong, I'm pretty sure we should `handle_verified_justification` here.

Reply:
Hmm, yeah, it won't hurt. It might even help.

Reply:
Although this might result in returning an error if peer's session number is too large.

Reply:
If it's way too large that sounds reasonable, if it's just a bit big this will trigger requesting more justifications.

Reply:
fixed